src/Input/Stepper.tsx

```diff
@@ -0,0 +1,257 @@
+import * as colors from "../colors";
+import { base } from "../typography";
+import { jsx } from "@emotion/core";
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+const IconPlus = require(`../../icons/IconPlus.tsx`).IconPlus;
+const IconMinus = require(`../../icons/IconMinus.tsx`).IconMinus;
+import { InputWrapper } from "./InputWrapper";
+
+const stepperWidth = 108;
+
+export type StepperLimit = "min" | "max";
+
+interface Props {
+  /**
+   * prefilled value
+   */
+  initialValue?: number;
+
+  /**
+   * accepts a number, not an event, since we only call onChange
+   * when the number should change, not when the user tries to change it
+   */
+  onChange: (value: number | undefined) => void;
+
+  /**
+   * callback when stepper is at min / max
+   */
+  onLimitReached?: (direction: StepperLimit) => void;
+  min?: number;
+  max?: number;
+  disabled?: boolean;
+
+  /**
+   * Visible description
+   */
+  description?: React.ReactNode;
+
+  /**
+   * Defaults to standard
+   */
+  size?: "small" | "standard" | "large";
+
+  /**
+   * Visible title
+   */
+  label?: React.ReactNode;
+
+  /**
+   * If this exists, it will be displayed as an error message
+   * below input box
+   */
+  error?: React.ReactNode;
+
+  /**
+   * If this exists, it will be displayed as an error message
+   * below input box
+   */
+  info?: React.ReactNode;
+}
+
+export const Stepper: React.FC<Props> = ({
+  initialValue,
+  onChange,
+  onLimitReached,
+  min = 0,
+  max = Number.MAX_SAFE_INTEGER,
+  disabled,
+  description,
+  label,
+  size = "standard",
+  error,
+  info,
+  ...otherProps
+}) => {
+  const [focused, setFocused] = useState<boolean>(false);
+  const [value, setValue] = useState<number>(initialValue || 10);
+  const atMax = value !== undefined && value >= max;
+  const atMin = value !== undefined && value <= min;
+
+  const handleChange = (newValue: number) => {
+    if (newValue !== undefined && newValue < min) {
+      setValue(min);
+      if (onLimitReached) onLimitReached("min");
+    } else if (newValue !== undefined && newValue > max) {
+      setValue(max);
+      if (onLimitReached) onLimitReached("max");
+    } else if (newValue !== value) {
+      setValue(newValue);
+    }
+    onChange(value);
+  };
+
+  const onIncrement = () => {
+    if (!atMax) {
+      handleChange((value || 0) + 1);
+    } else if (onLimitReached) {
+      onLimitReached("max");
+    }
+  };
+
+  const onDecrement = () => {
+    if (!atMin) {
+      handleChange((value || 0) - 1);
+    } else if (onLimitReached) {
+      onLimitReached("min");
+    }
+  };
+
+  const handleFocus = () => {
+    setFocused(true);
+  };
+
+  const handleBlur = () => {
+    setFocused(false);
+    handleChange(value);
+  };
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = parseInt(event.target.value, 10);
+    if (!isNaN(newValue)) {
+      handleChange(newValue);
+    } else {
+      handleChange(0);
```

Comment:
I don't think it makes sense to fire this with a 0 if the value is not a number.  `undefined` makes more sense to me.